injection/injection.go

```diff
@@ -59,10 +59,37 @@ func EnableInjectionOrDie(ctx context.Context, cfg *rest.Config) (context.Contex
 
 	ctx, informers := Default.SetupInformers(ctx, cfg)
 
+	includedInformers := make([]controller.Informer, 0)
+
+	for _, inf := range informers {
+		if p := getExcludeInformerPredicate(ctx); p != nil && (*p)(ctx, inf) {
+			continue
+		}
+		includedInformers = append(includedInformers, inf)
+	}
+
 	return ctx, func() {
-		logging.FromContext(ctx).Info("Starting informers...")
-		if err := controller.StartInformers(ctx.Done(), informers...); err != nil {
+		logging.FromContext(ctx).Infof("Starting %d informers...", len(includedInformers))
+		if err := controller.StartInformers(ctx.Done(), includedInformers...); err != nil {
 			logging.FromContext(ctx).Fatalw("Failed to start informers", zap.Error(err))
 		}
 	}
 }
+
+type ExcludeInformerPredicate func(ctx context.Context, inf controller.Informer) bool
+type excludeInformerPredicateKey struct{}
+
+// WithExcludeInformerPredicate sets the predicate to exclude informers from being started.
+// If predicate returns true the informer will not be started.
+func WithExcludeInformerPredicate(ctx context.Context, predicate ExcludeInformerPredicate) context.Context {
```

Comment:
Maybe "WithInformerExcludingFilter" or something in that direction?

Reply:
fixed